cfpb/consumerfinance.gov
# Handle localStorage logic in case it is blocked

If cookies are completely blocked, an error is thrown `Failed to read the 'localStorage' property from 'Window': Access is denied for this document.`

## Changes

- Utilize `web-storage-proxy` utility script where localStorage/sessionStorage is used.
- Try…catch to see if localStorage is available, and if not, gracefully fall back.

## How to test this PR

1. Go into Chrome settings via Settings > Privacy and Security > Cookies and other site data
2. Check "Block all cookies (not recommended)"
3. Visit https://www.consumerfinance.gov/owning-a-home/ and scroll down and see errors in dev console.
3. Pull branch and build and run locally. Visit http://localhost:8000/owning-a-home/ and scroll down and see that there are no errors in the dev console.

## Notes and todos

- The ["orange-explorer"](https://github.com/cfpb/consumerfinance.gov/blob/master/cfgov/hmda/templates/hmda/orange-explorer.html) has some direct references to localStorage too, but that page is a large inlined frankenstein monster I'm not going to touch in this PR. EDIT: This tool will be sunset [next month](https://github.com/cfpb/consumerfinance.gov/pull/5952)!
